File: tests/dynamic_fixtures/js/ssrf_negative.js
// SSRF — negative fixture.
// Safe: only allows http/https scheme; file:// and other schemes are rejected.
// Entry: fetchUrl(targetUrl)  Cap: SSRF
// Expected verdict: NotConfirmed

const http = require('http');
const https = require('https');
const url = require('url');

function fetchUrl(targetUrl) {
    const parsed = url.parse(targetUrl);
    if (parsed.protocol !== 'http:' && parsed.protocol !== 'https:') {
        process.stdout.write('Scheme not allowed: ' + parsed.protocol + '\n');
        return;
    }
    const mod = parsed.protocol === 'https:' ? https : http;
    const req = mod.get(targetUrl, (res) => {
        res.on('data', (chunk) => process.stdout.write(chunk.toString().substring(0, 64)));
    });
    req.on('error', () => {});
    req.end();
}

module.exports = { fetchUrl };

File: tests/dynamic_fixtures/java/ssrf_negative.java
// SSRF — negative fixture.
// Safe: only allows http/https scheme; file:// rejected.
// Entry: Entry.fetchUrl(String)  Cap: SSRF
// Expected verdict: NotConfirmed

import java.io.*;
import java.net.*;

public class Entry {
    public static void fetchUrl(String targetUrl) throws Exception {
        URL url = new URL(targetUrl);
        String proto = url.getProtocol();
        if (!proto.equals("http") && !proto.equals("https")) {
            System.out.println("Scheme not allowed: " + proto);
            return;
        }
        try (InputStream in = url.openStream();
             BufferedReader reader = new BufferedReader(new InputStreamReader(in))) {
            String line;
            while ((line = reader.readLine()) != null) {
                System.out.println(line.substring(0, Math.min(line.length(), 64)));
            }
        } catch (Exception e) {
            System.out.println("Connection error");
        }
    }
}
